subgraphs/compound-forks/protocols/bastion-protocol/src/mapping.ts

```diff
@@ -402,192 +448,206 @@ function getOrCreateProtocol(): LendingProtocol {
-      token.name = "Bastion Protocol";
-      token.symbol = "BSTN";
-      token.decimals = 18;
-    }
-    token.lastPriceUSD = getBastionPrice();
-    token.lastPriceBlockNumber = blockNumber;
-    token.save();
+}
 
-    supplyRewardToken = getOrCreateRewardToken(token, RewardTokenType.DEPOSIT);
+function updateRewardTokenPrices(market: Market, event: ethereum.Event): void {
+  if (!market.rewardTokens || market.rewardTokens!.length == 0) {
+    return;
   }
 
-  if (!trySupplyOne.reverted) {
-    if (
-      supplyRewardSpeed &&
-      supplyRewardSpeed.gt(BIGINT_ZERO) &&
-      trySupplyOne.value.gt(BIGINT_ZERO)
-    ) {
-      log.warning(
-        "[updateRewards] Multiple reward speeds found for supply side: {} {}",
-        [marketAddress.toHexString()]
-      );
-    }
-
-    if (trySupplyOne.value.gt(BIGINT_ZERO)) {
-      supplyRewardSpeed = trySupplyOne.value;
-
-      // load wNEAR token
-      token = Token.load(REWARD_TOKENS[INT_ONE].toHexString());
-      if (!token) {
-        // wNEAR is already made from the NEAR market
-        log.warning("[updateRewards] wNEAR not found: {}", [
-          REWARD_TOKENS[INT_ONE].toHexString(),
-        ]);
-        return;
-      }
-
-      supplyRewardToken = getOrCreateRewardToken(
-        token,
-        RewardTokenType.DEPOSIT
-      );
-    }
+  const rewardEmissions = market.rewardTokenEmissionsAmount;
+  if (!rewardEmissions || rewardEmissions.length == 0) {
+    return;
   }
+  const rewardEmissionsUSD: BigDecimal[] = [];
 
-  // if a supply side reward is successfully found, update rewards
-  if (supplyRewardSpeed && token) {
-    const priceUSD = token.lastPriceUSD ? token.lastPriceUSD : BIGDECIMAL_ZERO;
-    rewardTokens.push(supplyRewardToken!.id);
-    const dailySupplyRewards = supplyRewardSpeed.times(
-      BigInt.fromI32(SECONDS_PER_DAY)
-    );
-    rewardEmissions.push(dailySupplyRewards);
+  for (let i = 0; i < market.rewardTokens!.length; i++) {
+    const rewardTokenId = market.rewardTokens![i];
+    const rewardToken = RewardToken.load(rewardTokenId);
+    if (!rewardToken) {
+      log.error("[]reward token {} for market {} doesn't exist", [
+        rewardTokenId,
+        market.id,
+      ]);
+      return;
+    }
+    const token = getOrCreateToken(Address.fromString(rewardToken.token));
+    const priceUSD = getRewardTokenPrice(token);
     rewardEmissionsUSD.push(
-      supplyRewardSpeed
+      rewardEmissions[i]
         .toBigDecimal()
         .div(exponentToBigDecimal(token.decimals))
-        .times(priceUSD!)
+        .times(priceUSD)
     );
   }
-
-  market.rewardTokens = rewardTokens;
-  market.rewardTokenEmissionsAmount = rewardEmissions;
   market.rewardTokenEmissionsUSD = rewardEmissionsUSD;
   market.save();
+
+  getOrCreateMarketDailySnapshot(
+    market,
+    event.block.timestamp,
+    event.block.number
+  );
+  getOrCreateMarketHourlySnapshot(
+    market,
+    event.block.timestamp,
+    event.block.number
+  );
```

Comment:
This function is called from `handleAccrueInterest()` which already creates snapshots. I think this is unnecessary